### DISCUSSION TITLE
PlatformIO and AVR64DD28/32 ... "Unrecognized combination of flash size and chip type"

### DISCUSSION THREAD

**Discussion by tactical-snacks** [SpenceKonde/DxCore#345] [Q&A] [answered]
I'm seeing a compile error ... `#error "Unrecognized combination of flash size and chip type"` in PlatformIO when building an .

the platformio.ini is ...
```
[env:AVR64DD28]
platform = atmelmegaavr
board = AVR64DD28
framework = arduino
```

core and toolchain ...

```
PLATFORM: Atmel megaAVR (1.6.0) > AVR64DD28
HARDWARE: AVR64DD28 24MHz, 8KB RAM, 64KB Flash
PACKAGES:
 - framework-arduino-megaavr-dxcore @ 1.4.10
 - toolchain-atmelavr @ 3.70300.220127 (7.3.0)
```

if i change the DD to DA, compiles without error. I found a similar error with the DA on google but can't find it here. Error message attached.
[error_message_avr64dd28.txt](https://github.com/SpenceKonde/DxCore/files/9539404/error_message_avr64dd28.txt)

**Comment by ObviousInRetrospect** [comment, marked as answer]
DD support is known broken in the current version

**Comment by SpenceKonde** [comment]
AVR DD-series parts are only planned for support in 1.5.0 and later. 1.5.0 is not ready for release, several important parts need to be fixed in the core and the toolchain files that support it are still not ready, I'm currently trying to pick apart a the lib/gcc directory tree to separate into files that are platform specific and which don't change and files which are not platform specific and do get updated, as I'd forgotten completely about one of the most important branches of the directory tree. 

Then mTC 2.6.0 gets made available on board manager with the new toolchain, and most of what needs to be ported to DxCore. 

Then that stuff gets ported to DxCore

Then I see what else is broken and start fixing it.

**Reply by tactical-snacks** [reply]
thank you for the explanation. tried to mark this as the answer but discussions not giving me the option.

**Comment by SpenceKonde** [comment]
AVR DD-series parts are only planned for support in 1.5.0 and later. 1.5.0
is not ready for release, several important parts need to be fixed in the
core and the toolchain files that support it are still not ready, I'm
currently trying to pick apart a the lib/gcc directory tree to separate
into files that are platform specific and which don't change and files
which are not platform specific and do get updated, as I'd forgotten
completely about one of the most important branches of the directory tree.

Then mTC 2.6.0 gets made available on board manager with the new toolchain,
and most of what needs to be ported to DxCore.

Then that stuff gets ported to DxCore

Then I see what else is broken and start fixing it.

On Fri, Sep 9, 2022 at 10:19 PM tactical snacks ***@***.***>
wrote:

> I'm seeing a compile error ... #error "Unrecognized combination of flash
> size and chip type" in PlatformIO when building an .
>
> the platformio.ini is ...
>
> [env:AVR64DD28]
> platform = atmelmegaavr
> board = AVR64DD28
> framework = arduino
>
> core and toolchain ...
>
> PLATFORM: Atmel megaAVR (1.6.0) > AVR64DD28
> HARDWARE: AVR64DD28 24MHz, 8KB RAM, 64KB Flash
> PACKAGES:
>  - framework-arduino-megaavr-dxcore @ 1.4.10
>  - toolchain-atmelavr @ 3.70300.220127 (7.3.0)
>
> if i change the DD to DA, compiles without error. I found a similar error
> with the DA on google but can't find it here. Error message attached.
> error_message_avr64dd28.txt
> <https://github.com/SpenceKonde/DxCore/files/9539404/error_message_avr64dd28.txt>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/DxCore/discussions/345>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEWYTWUKZW5ZV44EPSXLV5PV27ANCNFSM6AAAAAAQJEII44>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***
